File: Practica/src/Guias/Guia4/EJ5_HTML/FormatText.java
package Guias.Guia4.EJ5_HTML;

public class FormatText implements HTMLText {

    private final HTMLText element;
    private final String format;

    public FormatText(HTMLText elem, String format) {
        this.element = elem;
        this.format = format;
    }

    @Override
    public String source() {
        return "<%s>%s<%s>".formatted(format, element.source(), format);
    }

    @Override
    public String toString() {
        return source();
    }
}

File: RUBY/Guias/Guia10/EJ1_html/format_text.rb
# frozen_string_literal: true
require_relative 'html_text'
require_relative 'plain_text'
class FormatText < PlainText
  def initialize(content, format)
    super(content)
    @format = format
  end
  def source
    "<#{@format}>#{super}<#{@format}"
  end
end
